cmd/snap-bootstrap/cmd_initramfs_mounts_test.go

```diff
@@ -283,725 +360,618 @@ recovery_system=20191118
 	c.Check(cloudInitDisable, testutil.FilePresent)
 }
 
-func (s *initramfsMountsSuite) TestInitramfsMountsRunModeStep1(c *C) {
+func (s *initramfsMountsSuite) TestInitramfsMountsRunModeHappy(c *C) {
 	s.mockProcCmdlineContent(c, "snapd_recovery_mode=run")
 
-	n := s.mockExpectedMountChecks(c,
-		notYetMounted{boot.InitramfsUbuntuBootDir},
-	)
-
-	// mock that we don't know which partition uuid the kernel was booted from
-	restore := main.MockPartitionUUIDForBootedKernelDisk("")
+	restore := s.mockSystemdMounts(c, []systemdMount{
+		ubuntuLabelMount("ubuntu-boot"),
+		ubuntuLabelMount("ubuntu-seed"),
+		ubuntuLabelMount("ubuntu-data"),
+		s.makeRunSnapSystemdMount(snap.TypeBase, s.core20),
+		s.makeRunSnapSystemdMount(snap.TypeKernel, s.kernel),
+	})
 	defer restore()
 
-	_, err := main.Parser().ParseArgs([]string{"initramfs-mounts"})
-	c.Assert(err, IsNil)
-	c.Assert(*n, Equals, 1)
-	c.Check(s.Stdout.String(), Equals, fmt.Sprintf(`/dev/disk/by-label/ubuntu-boot %[1]s/ubuntu-boot
-`, boot.InitramfsRunMntDir))
-}
-
-func (s *initramfsMountsSuite) TestInitramfsMountsRunModeStep1WithBootedKernelPartUUID(c *C) {
-	s.mockProcCmdlineContent(c, "snapd_recovery_mode=run")
-
-	n := s.mockExpectedMountChecks(c,
-		notYetMounted{boot.InitramfsUbuntuBootDir},
-	)
+	// mock a bootloader
+	bloader := boottest.MockUC20RunBootenv(bootloadertest.Mock("mock", c.MkDir()))
+	bootloader.Force(bloader)
+	defer bootloader.Force(nil)
 
-	restore := main.MockPartitionUUIDForBootedKernelDisk("specific-ubuntu-boot-partuuid")
+	// set the current kernel
+	restore = bloader.SetEnabledKernel(s.kernel)
 	defer restore()
 
-	_, err := main.Parser().ParseArgs([]string{"initramfs-mounts"})
-	c.Assert(err, IsNil)
-	c.Assert(*n, Equals, 1)
-	c.Check(s.Stdout.String(), Equals, fmt.Sprintf(`/dev/disk/by-partuuid/specific-ubuntu-boot-partuuid %[1]s/ubuntu-boot
-`, boot.InitramfsRunMntDir))
-}
-
-func (s *initramfsMountsSuite) TestInitramfsMountsRunModeStep2(c *C) {
-	s.mockProcCmdlineContent(c, "snapd_recovery_mode=run")
+	makeSnapFilesOnEarlyBootUbuntuData(c, s.kernel, s.core20)
 
-	n := s.mockExpectedMountChecks(c,
-		mounted{boot.InitramfsUbuntuBootDir},
-		notYetMounted{boot.InitramfsUbuntuSeedDir},
-	)
+	// write modeenv
+	modeEnv := boot.Modeenv{
+		Mode:           "run",
+		Base:           s.core20.Filename(),
+		CurrentKernels: []string{s.kernel.Filename()},
+	}
+	err := modeEnv.WriteTo(boot.InitramfsWritableDir)
+	c.Assert(err, IsNil)
 
-	_, err := main.Parser().ParseArgs([]string{"initramfs-mounts"})
+	_, err = main.Parser().ParseArgs([]string{"initramfs-mounts"})
 	c.Assert(err, IsNil)
-	c.Assert(*n, Equals, 2)
-	c.Check(s.Stdout.String(), Equals, fmt.Sprintf(`/dev/disk/by-label/ubuntu-seed %[1]s/ubuntu-seed
-`, boot.InitramfsRunMntDir))
 }
 
-func (s *initramfsMountsSuite) TestInitramfsMountsRunModeStep3(c *C) {
-	s.mockProcCmdlineContent(c, "snapd_recovery_mode=run")
+func (s *initramfsMountsSuite) TestInitramfsMountsInstallModeRealSystemdMountTimesOutNoMount(c *C) {
```

Comment:
this bit of behavior could be tested in isolation maybe, see comment about a file just with systemdMount and struct

Reply:
Since I already wrote these tests, let's just do both, I will split things out into another file with more tests just around doSystemdMount with `systemd-mount` mocked via `testutil.MockCommand`, and we can keep these "end-to-end" tests that also mock systemd-mount.

Reply:
actually I ended up breaking this out into a smaller PR that should be easier to review: https://github.com/snapcore/snapd/pull/9067